dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -1054,23 +1054,45 @@ public int countCollectionsWithSubmit(String q, Context context, Community commu
         return (int) resp.getTotalSearchResults();
     }
 
+
     @Override
     @SuppressWarnings("rawtypes")
     public List<Collection> findAllCollectionsByEntityType(Context context, String entityType)
-            throws SearchServiceException {
+            throws SQLException, SearchServiceException {
         List<Collection> collectionList = new ArrayList<>();
 
+        // we need to turn off authorization to retrieve all
+        // collections, not only the ones for the user in session
+        context.turnOffAuthorisationSystem();
+
+        //calculate the total number of collections
+        int collectionsNumber = countCollectionsWithSubmit(null, context, null, entityType);
+
+        int calculatedPages = (collectionsNumber / SOLR_ROWS_PER_PAGE) + 1;
+
         DiscoverQuery discoverQuery = new DiscoverQuery();
         discoverQuery.setDSpaceObjectFilter(IndexableCollection.TYPE);
-        discoverQuery.addFilterQueries("dspace.entity.type:" + entityType);
+        discoverQuery.setMaxResults(SOLR_ROWS_PER_PAGE);
+
+        for (int page = 1; page <= calculatedPages; page++) {
 
-        DiscoverResult discoverResult = searchService.search(context, discoverQuery);
-        List<IndexableObject> solrIndexableObjects = discoverResult.getIndexableObjects();
+            discoverQuery.setStart((page - 1) * SOLR_ROWS_PER_PAGE);
 
-        for (IndexableObject solrCollection : solrIndexableObjects) {
-            Collection c = ((IndexableCollection) solrCollection).getIndexedObject();
-            collectionList.add(c);
+            //retrieve the specific page with results
+            DiscoverResult discoverResult = retrieveCollectionsWithSubmit(context, discoverQuery,
```

Comment:
Overall, this PR appears to work.  But, I'm wondering why we aren't just calling `findCollectionsWithSubmit(null, context, null, entityType, -1, -1)`?   It appears that would immediately return all Collections in one List, without having to use the "for" loop that you have in this code?

If there's a reason why you prefer your approach, that's fine.  I'm just noting that it seems like your logic is the same as the logic would be in `findCollectionsWithSubmit()` if you disabled the offset & limt.